src/evict/evict_lru.c

```diff
@@ -421,13 +417,14 @@ __evict_server(WT_SESSION_IMPL *session, bool *did_work)
 
 	/* Eviction is stuck, check if we have made progress. */
 	if (*did_work) {
-#if defined(HAVE_DIAGNOSTIC) || defined(HAVE_VERBOSE)
-		__wt_epoch(session, &cache->stuck_time);
+#if !defined(HAVE_DIAGNOSTIC)
+		/* Need verbose check only if not in diagnostic build */
+		if (WT_VERBOSE_ISSET(session, WT_VERB_EVICT_STUCK))
 #endif
+			__wt_epoch(session, &cache->stuck_time);
```

Comment:
The function __evict_server gets called repeatedly for eviction pass. So having the __wt_epoch outside of diagnostic mode or in non-verbose mode might have performance impact on eviction thread. Has the eviction thread's performance been compared before and after this change?

Reply:
@nehakhatri5 We ran the Jenkins performance tests, which include a number that stress eviction and there was no noticeable change, so I think this one is safe.